Sources/Profile.php

```diff
@@ -922,13 +922,14 @@ function alerts_popup($memID)
 
 	// No funny business allowed
 	$counter = isset($_REQUEST['counter']) ? max(0, (int) $_REQUEST['counter']) : 0;
+	$limit = !empty($modSettings['alerts_per_page']) && (int) $modSettings['alerts_per_page'] < 1000 ? min((int) $modSettings['alerts_per_page'], 1000) : 25;
```

Comment:
$modSettings needs to be added to the global variables declaration in this function.

Reply:
My bad. I  tested the logic was working with a db_quote, but I didn't test it in all places.